src/main/java/org/madhuri/app/message/service/ChannelService.java

```diff
@@ -23,28 +26,41 @@ public void deleteChannel(long id) {
 		channelDAO.deleteChannel(id);
 	}
 
-	
+	public Channel updateChannel(Channel updatedChannel) {
+		if (updatedChannel.getId() == null) {
+			throw new IllegalArgumentException("Channel ID cannot be null");
+		}
 
-	public Channel getChannelById(long id) {
-		return channelDAO.getChannelById(id);
+		// Check if the channel exists
+		Channel existingChannel = channelDAO.getChannelById(updatedChannel.getId());
+		if (existingChannel == null) {
+			throw new NotFoundException("Channel with ID " + updatedChannel.getId() + " not found");
+
+		}
+
+		// Update the fields of the existing channel with the updated channel's data
+		existingChannel.setChannelName(updatedChannel.getChannelName());
+		existingChannel.setAdminId(updatedChannel.getAdminId());
+		existingChannel.setWelcomeMessage(updatedChannel.getWelcomeMessage());
+		existingChannel.setUpdatedAt(updatedChannel.getUpdatedAt());
+		existingChannel.setUpdatedBy(updatedChannel.getUpdatedBy());
+		
+		// Update user list in the existing channel
+	    List<User> updatedUserList = updatedChannel.getUsers();
```

Comment:
while updating channels, why are you updating the userList ? 
is users part of channel table ? if not then this should not happen here